### DISCUSSION TITLE
Prisma:warn Prisma failed to detect the Linux distro in use, and may not work as expected. Defaulting to "debian"

### DISCUSSION THREAD

**Discussion by DevLobato** [prisma/prisma#17380] [Q&A] [answered]
Hi, guys!

I'm using WSL2 setup, with ArchWSL [https://github.com/yuk7/ArchWSL](url) distro, and initiating tyscript, SQLite and React project.

When installing and running Prisma on project, a warning message appears:
"prisma:warn Prisma failed to detect the Linux distro in use, and may not work as expected. Defaulting to "debian"."

Anyone that uses ArchWSL/WSL2/Windows11 has the same issue?

I know it's just a warning, but I really don't know what impact it can have.

![image](https://user-images.githubusercontent.com/102265245/213038239-d6d4ca9b-ebbb-40bb-9e14-3cdef2d9fc8e.png)

Nodejs version: 18.13.0
NPM version: 8.19.3
Prisma version: 4.9.0
@prisma/client version: 4.9.0

Grateful a lot!

Vinicius.

**Comment by nurul3101** [comment, marked as answer]
Hey @DevLobato 👋 

I think the warning is because the particular Linux distro that you are using. Here's a [list](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#binarytargets-options) of all officially supported OS and distros.
Are you getting any errors when using prisma?

**Reply by pantharshit00** [reply]
You can use `PRISMA_DISABLE_WARNINGS=1` in the environment variables to suppress the warning if you don't want it in your logs. Please do keep in mind that it will disable all warnings thrown by Prisma.

**Reply by DevLobato** [reply]
Hi, @nurul3101!

No, i don't. Just warning message, and doesn't crash the application.
I'll continue to use, because Prisma helps me a lot.

@pantharshit00, thank you so much! I'll use carefully.

Thank you all guys!

**Reply by jkomyno** [reply]
Hi @DevLobato, on Arch Linux, that warning is nothing you need to worry about. Internally, we map the Linux distros we explicitly test for into either `debian`, `rhel`, or `musl`, to understand which Prisma Engines to download. We use the engines built for `debian` as a fallback. So, if `prisma` worked for you before `prisma@4.9.0`, it'll keep working after as well. In the latest release, we have made this fallback more transparent to the users, in an effort to help users understand what's going on when Prisma is installed on their system / or Docker, while also expanding the number of system configurations supported.

I hope this clarifies things a bit :)

**Reply by DevLobato** [reply]
Hi, @jkomyno!

It cleared even more. Like i said, i'll continue developing with Arch Linux, a distro that i liked much, and at the time, don't have any issue.

Thanks a lot.

Best regards,

Vinicius.

**Comment by kennyk-ing** [comment]
@jkomyno @pantharshit00,

I'm curious why this comes up in an Arch system when Arch is in that list of supported OS/distros.

Also, It would be nice if it was possible to disable just this specific warning without risk of missing other warnings.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
